java/src/main/java/org/hyperledger/fabric/client/FileCheckpointer.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2022 IBM All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.fabric.client;
+
+import com.google.gson.Gson;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+class CheckpointerState {
+    private long blockNumber;
+    private String transactionId;
+    CheckpointerState(final long blockNumber, final String transactionId) {
+        this.blockNumber = blockNumber;
+        this.transactionId = transactionId;
+    }
+
+    public String getTransactionId() {
+        return transactionId;
+    }
+
+    public long getBlockNumber() {
+        return blockNumber;
+    }
+}
+
+/**
+ * FileCheckpointer to store checkpointer state during file read write operations.
+ */
+public final class FileCheckpointer implements Checkpointer {
+
+    private long blockNumber;
+    private String transactionId = "";
+    private String path = "";
+
+    FileCheckpointer(final String path) throws Exception {
+        this.path = path;
+        this.init();
+    }
+
+    private void init() throws Exception {
+        this.loadFromFile();
+        this.saveToFile();
+    }
+
+    /**
+     * CheckpointBlock checkpoints the block.
+     * @param blockNumber
+     */
```

Comment:
For methods that provide an implementation for something defined in the interface, always use the `@Override` annotation so the compiler can check you are correctly overriding and not accidentally overloading the parent method definition. It also mean you pick up the parent Javadoc for free, so no need to repeat it here.
```suggestion
    @Override
```